explorer/ast/value_transform.h

```diff
@@ -114,13 +168,17 @@ class TransformBase {
   auto operator()(const llvm::StringMap<T>& map) -> llvm::StringMap<T> {
     llvm::StringMap<T> result;
     for (const auto& it : map) {
-      result.insert({Transform(it.first()), Transform(it.second)});
+      result.insert(
+          {TransformOrOriginal(it.first()), TransformOrOriginal(it.second)});
     }
     return result;
   }
 
  private:
   Nonnull<Arena*> arena_;
+  // Temporary storage for an error that was produced during transformation
+  // that has not yet been handed back to the caller.
+  std::optional<Error> error_;
```

Comment:
Can this be `ErrorOr<Success>`? I think that's the direct we'd chosen over std::optional<Error>

Reply:
Hah, I started with `ErrorOr<Success>`, but switched to this because it seemed to more clearly represent what we mean here -- temporary storage for an out-of-band error, rather than a status for the transformation. But I have no strong preference; switched back to `ErrorOr`.